File: library/src/main/java/org/incendo/interfaces/interfaces/PlayerInterface.java
package org.incendo.interfaces.interfaces;

import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.incendo.interfaces.click.ClickHandler;
import org.incendo.interfaces.pane.PlayerPane;
import org.incendo.interfaces.transform.AppliedTransform;
import org.incendo.interfaces.view.InterfaceView;
import org.incendo.interfaces.view.PlayerInterfaceView;

import java.util.Collection;
import java.util.Map;

public record PlayerInterface(
    Map<InventoryCloseEvent.Reason, CloseHandler> closeHandlers,
    Collection<AppliedTransform<PlayerPane>> transforms,
    Collection<ClickHandler> clickPreprocessors
) implements Interface<PlayerPane> {

    public static final int NUMBER_OF_COLUMNS = 9;

    @Override
    public int rows() {
        return 4;
    }

    @Override
    public PlayerPane createPane() {
        return new PlayerPane();
    }

    @Override
    public PlayerInterfaceView open(final Player player, final InterfaceView parent) {
        PlayerInterfaceView view = new PlayerInterfaceView(player, this, parent);
        view.open();

        return view;
    }

}

File: next/src/main/java/org/incendo/interfaces/next/interfaces/PlayerInterface.kt
package org.incendo.interfaces.next.interfaces

import org.bukkit.entity.Player
import org.incendo.interfaces.next.click.ClickHandler
import org.incendo.interfaces.next.pane.PlayerPane
import org.incendo.interfaces.next.transform.AppliedTransform
import org.incendo.interfaces.next.view.InterfaceView
import org.incendo.interfaces.next.view.PlayerInterfaceView

public class PlayerInterface internal constructor(
    override val transforms: Collection<AppliedTransform<PlayerPane>>,
    override val clickPreprocessors: Collection<ClickHandler>
) : Interface<PlayerPane> {

    public companion object {
        public const val NUMBER_OF_COLUMNS: Int = 9
    }

    override val rows: Int = 4

    override fun createPane(): PlayerPane = PlayerPane()

    override fun open(player: Player): InterfaceView<PlayerPane> {
        val view = PlayerInterfaceView(player, this)
        view.open()

        return view
    }
}
